### DISCUSSION TITLE
Dialogues

### DISCUSSION THREAD

**Discussion by julianogales** [SIGGgames/PacoMonkey#22] [Ideas]
There are two types of dialogues depending on the importance of the NPC:

**Secondary NPC**
If the NPC is not important or doesn't give valuable information it will be classified as a 'secondary NPC'.
- The dialog will pop up automatically as soon as the NPC appears on screen and it will remain until it's no longer visible.
> e.g. "It's so cold my balls are frozen."

**Main NPC**
When the NPC is an important character in the story or gives important information to the player it will be classified as a 'main NPC'.
- The dialog will require manual interaction with the NPC from the player. The camara and the main character will be frozen until the interaction is over or the player exits the conversation (can be done at any moment).
- The first interaction will announce what the information is about, without giving the actual info.
- On the following interactions (the number will deppend on the complexity of the given information), the NPC will ask something that the player is supposed to know from previous dialogs with other NPCs. Then the player will be required to choose 1 among 4 possible answers. If the player gives the wrong answer, the NPC will not be able to guide him or will lead the player through the wrong path.

> e.g.:
> - [PACOMONKEY]: "I'm looking for the monkey who killed my mom. Do you know anything about him?"
> - [MAIN NPC1]: "Maybe. If you give me his name I might know something about him."
> - _Choose between:_ {(Gorilla Manilla), (Abe the Ape), (MotherKiller123), (Zuzú)}
>   - Gives the correct answer: "I don't know where he is now but I remember him. He was always looking for The Great Banana Tree. If you find the Great Banana Tree you will find him."
>   - Gives wrong answer: "I'm sorry kiddo, I've never heard of him. Good luck."
>   - Gives another wrong answer: "Oh yes! I know him. He lives in the village where you just came from. You should turn and go now. Run before he gets away!"

- When you interact again with the NPC, if you gave him the wrong answer it will ask you again the same question. If you gave him the correct answer the dialog will change and it will give you a short version of the given clue.

_Before creating the dialogs, first we need a schematic guide of the world map with all the sorted biomes, along with the important NPCs._

**Comment by Abelitux** [comment]
I would make the secondary dialogs more opcional. I mean, you have to click the key that appears above the NPC to start the discussion. This is better for the player because if the dialog appears without his permission, he may feel weird and also the dialog occupies so much space. Also, I think it's better for the player not to be able to move during the conversation, for further problems. Anyways, the player will be able to skip the conversation.

**Reply by julianogales** [reply]
If we make the secondary dialogues smaller, without occupying so much space (like I mention in the next discussion), it wouldn't feel weird for the player when it pops-up. It would be more similar to a background object than to an interactive NPC and, therefore, the player could move freely.
But for the main NPCs case, I do believe that it would be better to freeze the player during the conversation to emphasize the importance of the dialogue and make him focus on the text.

**Comment by Abelitux** [comment]
Also, important to know if dialogues are just above the characters, or there are facial portraits dedicated to every character to know who is talking.
> e.g.:
> **Portrait dialogue:**
> ![dialogue portrait type](https://github.com/SIGGgames/PacoMonkey/assets/22718557/b9635e81-17b5-4b68-9e9d-f75c39110dd4)
> **Above-character dialogue:**
> ![above-character dialogues](https://github.com/SIGGgames/PacoMonkey/assets/22718557/04f314bf-aa92-4801-a5c4-44312a4e42a9)

**Reply by julianogales** [reply]
The original idea for the secondary NPCs was that the dialogue that automatically pops-up when the NPCs enters the screen is an 'above-character dialogue' like you mentioned, but not exactly like the example. The size would be smaller, without covering the entire screen, and wouldn't require interaction to continue or stop it, since every secondary NPC interaction only occupies one line. But I like the idea of putting the name of the speaker on top of the text.

> e.g.
> **Bubble Dialogue above NPC**:
> ![Xenoblade-Chronicles-3-review-conversation](https://github.com/SIGGgames/PacoMonkey/assets/150971528/343607f2-9ae6-40e4-8558-035609f19404)

And for the main NPCs I like the 'portrait dialogue' idea, but not as big as the sample you provided. If it's just dialogue, these two options would fit well:
> e.g.
> **Transparent Background with detailed image attached**:
> ![101e77b6-4394-4b98-9ea8-c02c2843ef7c](https://github.com/SIGGgames/PacoMonkey/assets/150971528/a08d5bb9-9a73-4a43-a0f4-51c073425bc2)
>
> **Simple dialogue box with simple drawing attached**:
> ![Captura de pantalla 2023-11-30 094308](https://github.com/SIGGgames/PacoMonkey/assets/150971528/8071b386-2f7f-49ac-b37c-74b90f827214)

If it's dialogue with answer required and options to choose, I like the idea of making the entire screen focus on the NPC, its dialogue and the provided answers.
> e.g.
> **Detailed NPC focused dialogue**:
> ![Captura de pantalla 2023-11-30 094733](https://github.com/SIGGgames/PacoMonkey/assets/150971528/ac440a42-0aae-4f2f-ab88-128f6b9fe4ef)

_But if you want to maintain a single uniform design for every NPC, we could choose/vote the best option that fits all cases._

**Comment by Aniol0012** [comment]
In my opinion, since it is a game where the camera is at a fixed point, the dialogues should be at a fixed point like in the lower central part. You could even consider adding cinematic-style black bars to add dialogue. This is my idea, what do you think?

![image](https://github.com/SIGGgames/PacoMonkey/assets/53788631/26b2ca84-8aee-426e-af45-3eb3657012a1)

**Reply by Abelitux** [reply]
It's not bad but the camera won't be at a fixed point, since it kinda follows the player.
